modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/side_panel/SidePanel.js

```diff
@@ -47,7 +47,7 @@ export default class SidePanel extends React.Component {
 			active: null,
 			closeButtonStyle: null,
 			currentURL: props.url || null,
-			iframeHandlerModalId: null,
+			iframeHandlerModalId: subscribeModal(),
```

Comment:
Why are we subscribing and unsubscribing modals? :thinking: What's the purpose of the `id` generated by the counter?

Reply:
The side panel component creates a modal instance which can be used within the iframe, il also has some logic which will eventually update the dataset where the action was originally triggered. 

Subscribing the modal makes the utilities in `/side_panel_content/index.js` aware that, in case the `OPEN_MODAL_FROM_IFRAME` event is triggered, only one modal will be open.

Actually this can be refactored by using our new Global Store mhh

Reply:
Something to have in mind is that we are in the process of migrating all existing modal to [Modal.js](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js). The new implementation does not have instances, modals are destroyed on close. The only way we can have multiple open modals is if they are nested, where we can cross-reference between them with `getOpener` and `getTop`. We cannot subscribe and reference later on, modals don't exist later on. When we come to the migration of commerce version of modal, we will need to either refactor commerce solution to work without subscription, or extend behavior of our standard util.

But that's all outside of scope of this task. For this PR, we can continue with the solution as it is.